### PR TITLE
general improvements

### LINKED ISSUE #12: General Improvements
I am raising this issue to discuss some general improvements to the photometry extension. For example,
- the photodetectors table has an attribute for peak wavelength, which I have previously used to specify the emission filter wavelength. However, it has come to my attention in the lerner conversion that the photodetector peak wavelength is not the same as the wavelength of the emission filter. To clear up this ambiguity, I propose that we add a fluorescence mini cube object which can store metadata information about the emission filter, dichroic mirror, and excitation filter (see [schematic](https://www.mightexbio.com/wp-content/uploads/2020/01/blogpost3_fig1_v2.jpg) by MightexBio).
- it may be useful for the fluorophores table to store the peak excitation wavelength and peak emission wavelength for each fluorophore.
- stereotactic coordinates for the fibers table
- DeconvolvedRoiResponseSeries --> DeconvolvedFiberPhotometryResponseSeries

### PR DESCRIPTION
**Pull request by luiztauffer**
- [x] emision and excitation lambda for fluorophores
- [x] stereotactic coordinates for the fibers table
- [x] DeconvolvedRoiResponseSeries --> DeconvolvedFiberPhotometryResponseSeries

### CODE REVIEW

**Comment by luiztauffer** [comment]
@pauladkisson named the new fluorophores attributes `excitation_lambda` and `emission_lambda` to keep it consistent with [Pynwb ophys module](https://pynwb.readthedocs.io/en/stable/pynwb.ophys.html)

File: src/spec/create_extension_spec.py [hunk 1]
```diff
@@ -63,6 +63,14 @@ def main():
                 shape=(None,),
                 neurodata_type_inc="VectorData",
             ),
+            NWBDatasetSpec(
+                name="coordinates",
+                doc="Fiber placement taxonomical coordinates in (AP, ML, Z) in mm relative to Bregma",
```
Comment (pauladkisson) [inline src/spec/create_extension_spec.py]:
```suggestion
                doc="Fiber placement in stereotactic coordinates (AP, ML, DV) mm relative to Bregma",
```

File: src/spec/create_extension_spec.py [hunk 2]
```diff
@@ -297,10 +276,34 @@ def main():
                 name="coordinates",
                 doc="injection taxonomical coordinates in (AP, ML, Z) in mm relative to Bregma",
```
Comment (pauladkisson) [inline src/spec/create_extension_spec.py]:
```suggestion
               doc="Fluorophore injection location in stereotactic coordinates (AP, ML, DV) mm relative to Bregma",
```

File: src/spec/create_extension_spec.py [hunk 3]
```diff
@@ -297,10 +276,34 @@ def main():
...
                 dtype="float",
-                shape=(None,3),
+                shape=(None, 3),
                 neurodata_type_inc="VectorData",
                 quantity="?",
             ),
+            NWBDatasetSpec(
+                name="emission_lambda",
+                doc="Emission wavelength of the fluorophore, in nanometers.",
```
Comment (pauladkisson) [inline src/spec/create_extension_spec.py]:
```suggestion
                name="emission_peak_wavelength",
                doc="Peak wavelength of emission of the fluorophore, in nanometers.",
```

File: src/spec/create_extension_spec.py [hunk 4]
```diff
@@ -297,10 +276,34 @@ def main():
...
+                neurodata_type_inc="VectorData",
+                attributes=[
+                    NWBAttributeSpec(
+                        name="unit", doc="wavelength unit", value="nanometers", dtype="text"
+                    )
+                ],
+            ),
+            NWBDatasetSpec(
+                name="excitation_lambda",
+                doc="Excitation wavelength of the fluorophore, in nanometers.",
```
Comment (pauladkisson) [inline src/spec/create_extension_spec.py]:
```suggestion
                name="excitation_peak_wavelength",
                doc="Peak wavelength of excitation of the fluorophore, in nanometers.",
```

File: src/pynwb/ndx_photometry/photometry.py [hunk 5]
```diff
@@ -5,6 +5,7 @@
 FluorophoresTable = get_class('FluorophoresTable', 'ndx-photometry')
 PhotodetectorsTable = get_class('PhotodetectorsTable', 'ndx-photometry')
 ExcitationSourcesTable = get_class('ExcitationSourcesTable', 'ndx-photometry')
+FiberPhotometryResponseSeries = get_class('FiberPhotometryResponseSeries', 'ndx-photometry')
```
Comment (pauladkisson) [inline src/pynwb/ndx_photometry/photometry.py:8]: Why does this need to be in the photometry.py file?

File: src/pynwb/tests/integration/test_photometry.py [hunk 6]
```diff
@@ -10,7 +10,7 @@
     FibersTable,
     PhotodetectorsTable,
     ExcitationSourcesTable,
-    DeconvolvedRoiResponseSeries,
+    DeconvolvedFiberPhotometryResponseSeries,
```
Comment (pauladkisson) [inline src/pynwb/tests/integration/test_photometry.py:13]: Looks like we need to update these tests -- probably best in a separate PR

**Review by pauladkisson — DISMISSED** [review]

**Comment by pauladkisson** [comment]
> named the new fluorophores attributes `excitation_lambda` and `emission_lambda` to keep it consistent with [Pynwb ophys module](https://pynwb.readthedocs.io/en/stable/pynwb.ophys.html)

Hmmm, I do think `emission_peak_wavelength` is a more informative name, and we are planning to propose an overhaul of microscopy in NWB (see [NEP Microscopy](https://docs.google.com/document/d/1ILo3sjBhIh_IXbqnqaqxSv_hF19HhRuhe4yrUH594Nk/edit#heading=h.o40x21ra4na6)).  But, maybe its best to keep it consistent with ophys for now...curious what @alessandratrapani thinks.

File: src/pynwb/tests/integration/test_photometry.py (hunk 6 above)
Comment (luiztauffer) [reply src/pynwb/tests/integration/test_photometry.py:13]: yes, better in a separate PR

File: src/pynwb/ndx_photometry/photometry.py (hunk 5 above)
Comment (luiztauffer) [reply src/pynwb/ndx_photometry/photometry.py:8]: oh, this was a test to see if I could resolve the current import errors... but that didn't help

**Comment by luiztauffer** [comment]
I agree that `emission_peak_wavelength` is a better name here, I'll be happy to change for that. But at the same time, it is the same property that's called `emission_lambda` in the current Ophys module, and it also seems to be the term used for NEP you shared... let's hear what @alessandratrapani and @weiglszonja think about it

**Comment by alessandratrapani** [comment]
> > named the new fluorophores attributes `excitation_lambda` and `emission_lambda` to keep it consistent with [Pynwb ophys module](https://pynwb.readthedocs.io/en/stable/pynwb.ophys.html)
> 
> Hmmm, I do think `emission_peak_wavelength` is a more informative name, and we are planning to propose an overhaul of microscopy in NWB (see [NEP Microscopy](https://docs.google.com/document/d/1ILo3sjBhIh_IXbqnqaqxSv_hF19HhRuhe4yrUH594Nk/edit#heading=h.o40x21ra4na6)). But, maybe its best to keep it consistent with ophys for now...curious what @alessandratrapani thinks.

I would definitely go with wavelength instead of lambda. The choice between `emission_peak` or just `emission` depends on what information on the emission spectrum we want to store. The `emission_peak_wavelength` is an intrinsic property of the fluorophore,  but the same fluorophore can be imaged through different optical filters that depend on the setup requirements (for example to avoid cross-talk with excitation wavelength or imaging multiple fluorophores). I would prefer to use `emission_wavelentgh` to store the emission wavelength that is read through the filter, since the emission peak wavelentgh can be easily retrieved from the indicator/opsin specified in the respective field. 
image

**Comment by luiztauffer** [comment]
thanks for the explanation @alessandratrapani 
I changed the fluorophores properties to `xxx_peak_wavelength`. We'll include filters information in a new minicube class, in a next PR.
If everyone agrees with the current state, I think we should merge this
@pauladkisson @weiglszonja

**Comment by alessandratrapani** [comment]
> thanks for the explanation @alessandratrapani I changed the fluorophores properties to `xxx_peak_wavelength`. We'll include filters information in a new minicube class, in a next PR. If everyone agrees with the current state, I think we should merge this @pauladkisson @weiglszonja

Sorry Luiz, I think I explain myself poorly. What I meant was I would not use `emission_peak_wavelentgh` because that is intrinsic in the fluorophore type, I don't think there is the need to specify it. I would go for `emission_wavelentgh` and report the emitted wavelength read through the filter, which is usually reported in the specific of the acquisition system. Same for the excitation wavelength.

But if you want to report only the fluorophore characteristics here, and the acquisition system in another place, then I guess it's ok, even if I think it's redundant

**Comment by luiztauffer** [comment]
oh I see what you mean now

> But if you want to characterize only the fluorophore characteristic here, and the acquisition system in another place, then I guess it's ok, even if I think it's redundant

yes, I feel it is important to have that property explicit within the Fluorophores items. We can then have the system's filter properties described in another class

File: README.md [hunk 7]
```diff
@@ -78,7 +78,9 @@ fluorophores_table = FluorophoresTable(description="fluorophores")
 fluorophores_table.add_row(
     label="dlight",
     location="VTA",
-    coordinates=(3.0,2.0,1.0)
+    coordinates=(3.0,2.0,1.0),
+    excitation_lambda=700.0,
+    emission_lambda=500.0
```
Comment (weiglszonja) [inline README.md]: I agree with keeping `emission_peak_wavelength` and `excitation_peak_wavelength` properties here, besides this small edit I think this PR is good to go. 

```suggestion
    excitation_peak_wavelength=700.0,
    emission_peak_wavelength=500.0
```

**Comment by luiztauffer** [comment]
someone needs to send an approving review for the merging to be allowed

**Review by weiglszonja — APPROVED** [review]
LGTM!

**Pull request merged**